instrumentation/netty/netty-4.1/testing/src/main/java/io/opentelemetry/instrumentation/netty/v4_1/AbstractNetty41ServerTest.java

```diff
@@ -0,0 +1,213 @@
+import static io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.NOT_FOUND;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.handler.codec.http.QueryStringDecoder;
+import io.netty.handler.logging.LogLevel;
+import io.netty.handler.logging.LoggingHandler;
+import io.netty.util.CharsetUtil;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerTestOptions;
+import io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint;
+import java.net.URI;
+import java.util.function.Supplier;
+
+public abstract class AbstractNetty41ServerTest extends AbstractHttpServerTest<EventLoopGroup> {
+
+  static final LoggingHandler LOGGING_HANDLER =
+      new LoggingHandler(AbstractNetty41ServerTest.class, LogLevel.DEBUG);
+
+  protected abstract void configurePipeline(ChannelPipeline channelPipeline);
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    options.setTestException(false);
+    // etty instrumentation collects route but doesn't use it for span names
+    options.setExpectedHttpRoute(unused -> null);
+    options.setExpectedServerSpanNameMapper((endpoint, method) -> "HTTP " + method);
+  }
+
+  @Override
+  protected EventLoopGroup setupServer() {
+    NioEventLoopGroup eventLoopGroup = new NioEventLoopGroup();
+
+    ServerBootstrap bootstrap =
+        new ServerBootstrap()
+            .group(eventLoopGroup)
+            .handler(LOGGING_HANDLER)
+            .childHandler(
+                new ChannelInitializer<SocketChannel>() {
+                  @Override
+                  protected void initChannel(SocketChannel socketChannel) {
+                    ChannelPipeline pipeline = socketChannel.pipeline();
+                    pipeline.addFirst("logger", LOGGING_HANDLER);
+                    pipeline.addLast(new HttpServerCodec());
+                    pipeline.addLast(
+                        new SimpleChannelInboundHandler<Object>() {
```

Comment:
Nit: WDYT of extracting this to a separate class? Just for better readability (less nesting/indentation)

Reply:
Yeah, I lifted this from a groovy test with minimal changes, but was thinking the same thing. Let's do a static inner class so its still encapsulated in the same class file.